### ISSUE TITLE
Using {{}} interpolation in an expression causes toUpperCase error

### ISSUE THREAD

**Issue by alxhub** [angular/angular#27740]
Accidental use of interpolation expressions inside an existing expression context results in a strange error.

Example:

```html
<div *ngIf="name && {{name}}"></div>
```

results in:

```text
Template parse errors:
TypeError: Cannot read property 'toUpperCase' of undefined ("<div [ERROR ->]*ngIf="name && {{name}}">
```

Live example here: https://stackblitz.com/edit/touppercase-of-undefined?file=src/app/app.component.html

Instead of throwing a more useful syntax error, we crash (probably in the attempt to throw the useful error).

**Comment by petebacondarwin** [comment]
This is being tracked in the aggregate issue of #43485 for which we need to create a project proposal.

**Comment by JoostK** [comment]
Fixed in #30300
